### PR TITLE
Check in/39194/dob login

### PR DESCRIPTION
**Pull request by brianseek**
## Description
Adds DOB validation behind feature flag. 

## Original issue(s)
department-of-veterans-affairs/va.gov-team#39194

## Testing done
Updated unit tests and created two new cypress tests for logging in with DOB.

## Screenshots
Screen Shot 2022-04-08 at 4 36 56 PM

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by jhonny218 — APPROVED** [review]

File: src/applications/check-in/api/local-mock-api/index.js [hunk 1]
```diff
@@ -75,6 +94,20 @@ const responses = {
     }
     return res.json(checkInData.post.createMockSuccessResponse({}));
   },
+  // v3
```
Comment (acrollet) [inline src/applications/check-in/api/local-mock-api/index.js:97]: Are we leaving this here intentionally?

File: src/applications/check-in/components/pages/validate/ValidateDisplay.unit.spec.jsx [hunk 2]
```diff
@@ -1,18 +1,51 @@
...
+          check_in_experience_lorota_security_updates_enabled: false,
+        },
+      };
+      beforeEach(() => {
+        store = mockStore(initState);
+      });
       it('passes axeCheck', () => {
-        axeCheck(<ValidateDisplay />);
+        axeCheck(
+          <Provider store={store}>
```
Comment (acrollet) [inline src/applications/check-in/components/pages/validate/ValidateDisplay.unit.spec.jsx:34]: Could we add a test component to provide this boilerplate? I18AwareProvider or some such?

File: src/applications/check-in/components/pages/validate/ValidateDisplay.unit.spec.jsx [hunk 3]
```diff
@@ -70,16 +139,68 @@ describe('check-in experience', () => {
...
+          };
+          const { getByTestId } = render(
+            <Provider store={mockStore(updatedStore)}>
+              <I18nextProvider i18n={i18n}>
+                <ValidateDisplay
+                  dobInput={{
+                    dob: {
+                      day: {
+                        value: 1,
+                        dirty: false,
```
Comment (acrollet) [inline src/applications/check-in/components/pages/validate/ValidateDisplay.unit.spec.jsx:172]: This is me being too lazy to look at the dob component but do we need any tests around what happens if `dirty` is true? Also I'd like to see a case with an ambiguous date like 3/12.

File: src/applications/check-in/utils/formatters/index.js [hunk 4]
```diff
@@ -24,4 +24,14 @@ const formatDemographicString = demographicString => {
   return demographicString;
 };
 
-export { formatPhone, formatDemographicString };
+/**
+ * @param {object} dateObject
+ */
+const formatDateObjectTo8601 = dateObject => {
+  return `${dateObject.year.value}-${String(dateObject.month.value).padStart(
```
Comment (acrollet) [inline src/applications/check-in/utils/formatters/index.js:31]: Could we use https://date-fns.org/v2.28.0/docs/formatISO with `{ representation: 'date' }` for this?

**Review by acrollet — CHANGES_REQUESTED** [review]
I have a few comments but overall this looks great and it's nice to be ahead of the game on this. Nice work! :-)

File: src/applications/check-in/components/pages/validate/ValidateDisplay.unit.spec.jsx (hunk 2 above)
Comment (brianseek) [reply src/applications/check-in/components/pages/validate/ValidateDisplay.unit.spec.jsx:34]: Will revist in next phase of translation

File: src/applications/check-in/components/pages/validate/ValidateDisplay.unit.spec.jsx (hunk 3 above)
Comment (brianseek) [reply src/applications/check-in/components/pages/validate/ValidateDisplay.unit.spec.jsx:172]: Will add a test in e2e for error state.

File: src/applications/check-in/utils/formatters/index.js (hunk 4 above)
Comment (brianseek) [reply src/applications/check-in/utils/formatters/index.js:31]: Change to extractDateFromVaDateComponent

Comment (brianseek) [reply src/applications/check-in/utils/formatters/index.js:31]: Add comment in doc block

File: src/applications/check-in/api/local-mock-api/index.js (hunk 1 above)
Comment (brianseek) [reply src/applications/check-in/api/local-mock-api/index.js:97]: forgot it! thanks for catching

File: src/applications/check-in/day-of/pages/ValidateVeteran.jsx [hunk 5]
```diff
@@ -52,7 +75,50 @@ const ValidateVeteran = props => {
     async () => {
       setLastNameErrorMessage();
       setLast4ErrorMessage();
-      if (!lastName || !last4Ssn) {
+      setDobErrorMessage();
+      if (isLorotaSecurityUpdatesEnabled) {
```
[the remark this answers is not in the document]
Comment (brianseek) [reply src/applications/check-in/day-of/pages/ValidateVeteran.jsx]: Refactored to use a shared util for both CI and PCI

**Review by acrollet — COMMENTED** [review]
argh, I never submitted these comments

**Comment by acrollet** [comment]
This is looking good and working well! Let's not have a default date if possible, or ask UX what date to use if the date component requires a set default.

**Comment by loripusey** [comment]
Since the person is entering their own DOB, I'm not sure why we would have a default value (unless the control requires a default).

**Review by acrollet — APPROVED** [review]
ship it!

**Pull request merged**